src/extension/background-script/CryptoServices/decryptFrom.ts

```diff
@@ -312,13 +314,43 @@ export async function* decryptFromMessageWithProgress_raw(
     return makeError(i18n.t('service_unknown_payload'))
 }
 
-export const decryptFromMessageWithProgress = memorizeAsyncGenerator(
-    decryptFromMessageWithProgress_raw,
+async function* decryptFromImageWithProgress_raw(
+    attachment: PostInfoImageAttachment,
+    author: ProfileIdentifier,
+    whoAmI: ProfileIdentifier,
+    publicShared: boolean,
+): ReturnOfDecryptPostContentWithProgress {
+    if (successDecryptionCache.has(attachment.url)) return successDecryptionCache.get(attachment.url)!
+    yield makeProgress('init')
+    const post = await decodeImageUrl(attachment.url, {
```

Comment:
What if the image is not downloadable in the background? maybe we need to try to download it in the content script cause it share the same domain as the SNS

Reply:
For now, we have no problem about this. Download in the background is on purpose it can avoid transferring a big chunk of data between pages.